src/Adapter/Geolocation/GeolocationByIpAddressConfiguration.php

```diff
@@ -26,55 +26,47 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Geolocation;
 
-use PrestaShop\PrestaShop\Adapter\Configuration;
-use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Configuration\AbstractMultistoreConfiguration;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Class GeolocationByIpAddressConfiguration is responsible for configuring geolocation configuration.
  */
-final class GeolocationByIpAddressConfiguration implements DataConfigurationInterface
+final class GeolocationByIpAddressConfiguration extends AbstractMultistoreConfiguration
 {
-    /**
-     * @var Configuration
-     */
-    private $configuration;
-
-    /**
-     * @param Configuration $configuration
-     */
-    public function __construct(
-        Configuration $configuration
-    ) {
-        $this->configuration = $configuration;
-    }
+    private const CONFIGURATION_FIELDS = ['geolocation_enabled'];
```

Comment:
What about constants to reuse in the entire class ? easier to maintain :-)
```suggestion
    private const CONFIGURATION_KEY = 'PS_GEOLOCATION_ENABLED';
    private const CONFIGURATION_FIELD = 'geolocation_enabled';
    private const CONFIGURATION_FIELDS = [self::CONFIGURATION_FIELD];
```
```suggestion
    public function getConfiguration()
    {
        return [
            self::CONFIGURATION_FIELD => (bool) $this->configuration->get(self::CONFIGURATION_KEY, false, $this->getShopConstraint()),
        ];
    }

    public function updateConfiguration(array $configuration)
    {
        if ($this->validateConfiguration($configuration)) {
            $this->updateConfigurationValue(self::CONFIGURATION_KEY, self::CONFIGURATION_FIELD, $configuration, $this->getShopConstraint());
        }

        return [];
    }

    protected function buildResolver(): OptionsResolver
    {
        $resolver = (new OptionsResolver())
            ->setDefined(self::CONFIGURATION_FIELDS)
            ->setAllowedTypes(self::CONFIGURATION_FIELD, 'bool');

        return $resolver;
    }
```

Reply:
Thanks, I'll look into it when the PR is no longer blocked by : https://github.com/PrestaShop/PrestaShop/issues/27540

Regards

Reply:
I've commented only 1 class, but my suggestion applies to all classes... not an absolute requirement though :)

Regards

Reply:
I will also look at my already merged PRs